app/views/courses/TeacherCoursesView.coffee

```diff
@@ -66,4 +68,10 @@ module.exports = class TeacherCoursesView extends RootView
     language = form.find('.language-select').val()
     window.tracker?.trackEvent 'Classes Guides Play Level', category: 'Teachers', courseID: courseID, language: language, levelSlug: levelSlug, ['Mixpanel']
     url = "/play/level/#{levelSlug}?course=#{courseID}&codeLanguage=#{language}"
-    application.router.navigate(url, { trigger: true })
+    firstLevelSlug = @campaigns.get(@courses.at(0).get('campaignID')).getLevels().at(0).get('slug')
+    if levelSlug is firstLevelSlug
+      @listenTo @openModalView(@heroSelectModal),
+        'hide': ->
+          application.router.navigate(url, { trigger: true })
+    else
+      application.router.navigate(url, { trigger: true })
```

Comment:
Didn't quite get tests working when doing it this way, but this is nice. We'll have to investigate getting tests working properly when using promises.